### PR TITLE
disable NodeDataMsg and snap Protocol

### PR DESCRIPTION
**Pull request by lightsing**
Fast sync uses NodeDataMsg in ETH66, disable it to reject all fast sync request.
Disable snap protocol to remove snap support from master node.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202810744511099

### CODE REVIEW

**Comment by lispc** [comment]
i suggest allow deadcode in CI in fact..    since i think "merge friendliness with upstream" is prior than "lint"

**Review by lispc — APPROVED** [review]

File: eth/backend.go [hunk 1]
```diff
@@ -525,7 +524,8 @@ func (s *Ethereum) BloomIndexer() *core.ChainIndexer   { return s.bloomIndexer }
 func (s *Ethereum) Protocols() []p2p.Protocol {
 	protos := eth.MakeProtocols((*ethHandler)(s.handler), s.networkID, s.ethDialCandidates)
 	if s.config.SnapshotCache > 0 {
```
Comment (Thegaram) [inline eth/backend.go:526]: If we want to disable `SnapshotCache`, we should probably do that here: https://github.com/scroll-tech/go-ethereum/blob/zkrollup/eth/backend.go#L102

File: eth/protocols/eth/handler.go [hunk 2]
```diff
@@ -180,12 +180,12 @@ var eth66 = map[uint64]msgHandler{
...
 	GetBlockBodiesMsg:             handleGetBlockBodies66,
 	BlockBodiesMsg:                handleBlockBodies66,
-	GetNodeDataMsg:                handleGetNodeData66,
-	NodeDataMsg:                   handleNodeData66,
-	GetReceiptsMsg:                handleGetReceipts66,
-	ReceiptsMsg:                   handleReceipts66,
-	GetPooledTransactionsMsg:      handleGetPooledTransactions66,
-	PooledTransactionsMsg:         handlePooledTransactions66,
+	//GetNodeDataMsg:                handleGetNodeData66,
+	//NodeDataMsg:                   handleNodeData66,
```
Comment (Thegaram) [inline eth/protocols/eth/handler.go:184]: It seems that we will disconnect nodes who send these, is that intentional?

Comment (lispc) [reply eth/protocols/eth/handler.go:184]: yes. since we use our own trie design, we cannot support snapshot/snapsync yet.

**Comment by Thegaram** [comment]
Closing in favour of #239.

**Pull request closed without merging**